### DISCUSSION TITLE
[Bug]: Weird red boxes around certain letters in RTL language

### DISCUSSION THREAD

**Discussion by aghasemi** [coder/code-server#4952] [Q&A] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### OS/Web Information

- Web Browser:
- Local OS:
- Remote OS:
- Remote Architecture:
- `code-server --version`:

### Steps to Reproduce

When writing code/strings in Persian (an RTL language using Perso-arabic alphabet), I see red boxes around certain letters. These boxes appear for both light and dark themes, and various fonts that I checked. See the image below for an example.

[
![cs](https://user-images.githubusercontent.com/7351535/156836158-acc2f195-399d-411d-8df5-5da36020cf44.jpeg)
](URL)

Any idea where this comes from?

### Expected

Persian text rendered the same way as English text. 

### Actual

There are red boxes around certain letters in text. You can try this text in the editor (e.g. Python).

```
'کلمه‌های خط‌ خورده در مصرع نیستند. کلمه‌های زرد رنگ در مصرع به کار رفته‌اند اما در مکان دیگری، و کلمه‌های پررنگ سبز در مصرع و در جای درست حضور دارند.'
```

### Does this issue happen in VS Code?

- [X] I cannot reproduce this in VS Code.

### Are you accessing code-server over HTTPS?

- [X] I am using HTTPS.

**Comment by jsjoeio** [comment]
No idea what this could be. Moved to a Q and A since the bug report is incomplete. If you can provide repro steps, happy to try locally

**Reply by aghasemi** [reply]
To reproduce, you can just create a new file or open an existing one, and write this line in it:

```
'کلمه‌های خط‌ خورده در مصرع نیستند. کلمه‌های زرد رنگ در مصرع به کار رفته‌اند اما در مکان دیگری، و کلمه‌های پررنگ سبز در مصرع و در جای درست حضور دارند.'

```

**Comment by lg-nineoneninethreenine** [comment]
Hmm... Urdu/Arabic, eh? Could be settings, there was another discussion with Russian about this.

**Reply by aghasemi** [reply]
Urdu and Arabic use (almost) the same writing system, yes.

**Reply by lg-nineoneninethreenine** [reply]
Try checking the settings for characters. I'll post the Russian language issue here, one min...

**Comment by lg-nineoneninethreenine** [comment, marked as answer]
Could be ambiguous chars: https://github.com/coder/code-server/discussions/4720
Guess that'd resolve this.

**Reply by aghasemi** [reply]
Many thanks. That was indeed the case, although for me it was invisible characters, not ambiguous ones.

By the way, there is a pretty painful RTL support, FYI, but I suspect it is VSCode's fault, not the code server.

Many thanks for the great effort, again.

**Reply by lg-nineoneninethreenine** [reply]
Thanks 😁
